tools/windows-node-installer/test/e2e/azure_test.go

```diff
@@ -0,0 +1,197 @@
+	winRmPriority = 300
+)
+
+// azureProvider stores Azure clients and resourceGroupName to access the windows node.
+type azureProvider struct {
+	// resourceGroupName of the Windows node
+	resourceGroupName string
+	// network security group client to check if winRmHttps port is opened or not.
+	nsgClient network.SecurityGroupsClient
+	// vm client to check for ansible ping on the windows node.
+	vmClient compute.VirtualMachinesClient
+}
+
+var (
+	// get the azure Auth Credentials specified from the env variable.
+	azureCredentials = os.Getenv("AZURE_AUTH_LOCATION")
+	// variable to be used for running the tests
+	azureInfo = azureProvider{}
+	// instanceIDs that are obtained from the windows-node-installer.json
+	instanceIDs []string
+	// secGroupIDs that are obtained from the windows-node-installer.json
+	secGroupsIDs []string
+)
+
+// TestWinRMSetup runs two tests i.e checks if the winRMHttps port is opened or not and
+// other test does an ansible ping check to confirm that windows node is correctly
+// configured to execute the remote ansible commands.
+func TestWinRMSetup(t *testing.T) {
+	setupAzureClients(t)
+	readInstallerInfo(t)
+	t.Run("check if WinRmHttps port is opened in the inbound security group rules list", testWinRmPort)
+	t.Run("check if ansible is able to ping on the WinRmHttps port", testAnsiblePing)
+}
+
+// checkForNil is a helper functions which checks if the object is a nil pointer or not.
+func checkForNil(v interface{}) bool {
+	return v == nil || (reflect.ValueOf(v).Kind() == reflect.Ptr &&
+		reflect.ValueOf(v).IsNil())
+}
+
+// setupAzureClients basically initializes the azure clients to be used in the tests.
+func setupAzureClients(t *testing.T) {
+	oc, err := client.GetOpenShift(kubeconfig)
+	require.NoError(t, err, "failed to initialize OpenShift client")
+	provider, err := oc.GetCloudProvider()
+	require.NoError(t, err, "failed to get cloud provider information")
+	resourceAuthorizer, err := auth.NewAuthorizerFromFileWithResource(azure.PublicCloud.ResourceManagerEndpoint)
+	require.NoError(t, err, "failed to get azure authorization token")
+	getFileSettings, err := auth.GetSettingsFromFile()
+	require.NoError(t, err, "failed to get info from AZURE_AUTH_LOCATION")
+	subscriptionId := getFileSettings.GetSubscriptionID()
+	vmClient := compute.NewVirtualMachinesClient(subscriptionId)
+	vmClient.Authorizer = resourceAuthorizer
+	nsgClient := network.NewSecurityGroupsClient(subscriptionId)
+	nsgClient.Authorizer = resourceAuthorizer
+	azureInfo.resourceGroupName = provider.Azure.ResourceGroupName
+	azureInfo.vmClient = vmClient
+	azureInfo.nsgClient = nsgClient
+}
+
+// readInstallerInfo reads the instanceIDs and secGroupsIDs from the
+// windows-node-installer.json file specified in "dir".
+func readInstallerInfo(t *testing.T) {
+	filePath := dir + "/windows-node-installer.json"
+	installerInfo, err := resource.ReadInstallerInfo(filePath)
+	require.NoError(t, err, "failed to read from ARTIFACT_DIR")
```

Comment:
failed to get installer info

Reply:
Or `failed to read installer info from %s, ARTIFACT_DIR`